analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -103,27 +108,46 @@ private void AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysis
 
         protected class RuleFactory
         {
-            public Type Type { get; }
             private readonly Func<SymbolicRuleCheck> createInstance;
+            private readonly Func<object> sonarFallbackFactory;
+
+            public Type Type { get; }
 
-            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance)
+            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance, Func<object> sonarFallbackFactory)
             {
                 Type = type;
                 this.createInstance = createInstance;
+                this.sonarFallbackFactory = sonarFallbackFactory;
             }
 
-            public SymbolicRuleCheck CreateInstance(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
+            public SymbolicRuleCheck CreateInstance(IAnalyzerConfiguration configuration, SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
             {
+                if (configuration.ForceSonarCfg && sonarFallbackFactory != null)
+                {
+                    return null;
+                }
+
                 var ret = createInstance();
                 ret.Init(sonarContext, nodeContext);
                 return ret;
             }
+
+            public object CreateSonarFallback(IAnalyzerConfiguration configuration) =>
+                configuration.ForceSonarCfg ? sonarFallbackFactory?.Invoke() : null;
+        }
+
+        protected class RuleFactory<TCheck> : RuleFactory
+            where TCheck : SymbolicRuleCheck, new()
+        {
+            public RuleFactory() : this(null) { }
+            public RuleFactory(Func<object> sonarFallbackFactory) : base(typeof(TCheck), () => new TCheck(), sonarFallbackFactory) { }
         }
 
-        private sealed class RuleFactory<TCheck> : RuleFactory
+        protected class RuleFactory<TCheck, TSonarFallback> : RuleFactory<TCheck>
             where TCheck : SymbolicRuleCheck, new()
+            where TSonarFallback : new()
```

Comment:
It is probably a good idea to constrain this type parameter to some "marker" interface (and ISymbolicExecutionAnalyzer should derive from it). Otherwise, the "Roslyn" rule can be passed here, which is an easy mistake to do (same class name!).